app/shop/pages/manageProducts/create.tsx

```diff
@@ -1,70 +1,44 @@
-import { BlitzPage, Routes, useMutation } from 'blitz'
+import { BlitzPage, Router, Routes, useMutation } from 'blitz'
+import { Suspense } from 'react'
 import { z } from 'zod'
 
-import Form, { FORM_ERROR } from 'app/core/components/Form'
-import LabeledTextField from 'app/core/components/LabeledTextField'
+import { Spinner } from 'app/core/components/Spinner'
 import { TopBar } from 'app/core/components/TopBar'
 import { useGoBack } from 'app/core/hooks/useGoBack'
 import { setupAuthRedirect } from 'app/core/utils/setupAuthRedirect'
 import { setupLayout } from 'app/core/utils/setupLayout'
 import createProduct from 'app/product/mutations/createProduct'
-import { CreateProduct } from 'app/product/validations'
-import { UploadImagesBlock } from 'app/shop/components/UploadImagesBlock'
+import { ProductFormValues } from 'app/product/validations'
+import { ProductForm } from 'app/shop/components/ProductForm'
 
 const CreateProductPage: BlitzPage = () => {
-  const [createProductMutation] = useMutation(createProduct)
-  const goBack = useGoBack(Routes.ManageProductsPage().pathname)
-
   return (
     <div>
       <TopBar
         backHref={Routes.ManageProductsPage().pathname}
         title="Add Product"
       />
-      <Form
-        className="py-3 px-6 flex flex-col gap-6"
-        submitText="Save"
-        schema={CreateProduct}
-        onSubmit={async (values: z.infer<typeof CreateProduct>) => {
-          try {
-            await createProductMutation(values)
-            await goBack()
-          } catch (error: any) {
-            return { [FORM_ERROR]: error.toString() }
-          }
-        }}
-      >
-        <LabeledTextField name="name" label="Product name" />
-        <LabeledTextField
-          name="price"
-          label="Price per item (฿)"
-          type="number"
-          step="0.01"
-        />
-        <LabeledTextField
-          name="stock"
-          label="Number of items in stock"
-          type="number"
-        />
-        <LabeledTextField
-          name="hashtags"
-          label="Hashtags"
-          asTextArea
-          style={{ height: 84 }}
-          caption="Separate each tag with a comma e.g. Shoes, Bag, Women, Accessories"
-        />
-        <LabeledTextField
-          name="description"
-          label="Product description"
-          asTextArea
-          style={{ height: 144 }}
-        />
-        <UploadImagesBlock />
-      </Form>
+      <Suspense fallback={<Spinner />}>
+        <CreateProductForm />
+      </Suspense>
     </div>
   )
 }
 
+const CreateProductForm = () => {
+  const [createProductMutation] = useMutation(createProduct)
+  const goBack = useGoBack(Routes.ManageProductsPage().pathname)
+  return (
+    <ProductForm
+      onSubmit={async (values: z.infer<typeof ProductFormValues>) => {
+        await createProductMutation(values)
+        await goBack()
+        Router.reload()
```

Comment:
Why reload?

Reply:
I have a problem updating state when a product is created (trying to avoid context because it seems overkill, but in this situation the `create.tsx` page is on a totally different page than `index.tsx`)

Reply:
Perhaps using `refetch()` will also fix this

Reply:
You can also use `getServerSideProps` in the list page. It will automatically reload when page is changed, and doesn't have loading states.

Reply:
[Here's an example](https://github.com/2110336-SoftwareEngineering2-2021-2/sec33_jim_carry/blob/3b9697602cfa1aea8296a740e319c583fb46d0c3/app/users/pages/addresses/index.tsx#L58)